### PR TITLE
Julia/feat/edit text files

### PR DESCRIPTION
**Pull request by jmelnych**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/redux/initialState.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ const initialAppState: AppState = {
     searchHistory: electronStore.get('appConfig.recentSearch') || [],
     currentMatch: null,
   },
+  supportedExtensions: ['.txt', '.md', '.html', '.json'],
```
Comment (jmelnych) [inline src/redux/initialState.ts:75]: these are the basic extensions I can see we might want to edit, please advice any other we can start with?

Comment (jmelnych) [inline src/redux/initialState.ts:75]: these are the basic extensions I can see we might want to edit, please advice any other we can start with?

Comment (jmelnych) [inline src/redux/initialState.ts:75]: these are the basic extensions I can see we might want to edit, please advice any other we can start with?

File: src/redux/selectors.ts [hunk 2]
```diff
@@ -166,6 +166,13 @@ export const fileIncludesSelector = createSelector(
   }
 );
 
+export const supportedExtensionsSelector = createSelector(
```
Comment (topliceanurazvan) [inline src/redux/selectors.ts:169]: Do we need a selector if we are returning directly the `supportedExtensions` from the state? 🤔

**Review by topliceanurazvan — CHANGES_REQUESTED** [review]
Looks good 👍🏻 

Noticed that the files that we are now supporting to edit are still marked as `unsupported` in the file explorer, even though we are able to click them. We should not mark them as unsupported anymore I think

image

image

File: src/redux/initialState.ts (hunk 1 above)
Comment (devcatalin) [reply src/redux/initialState.ts:75]: we should include all text files extensions from here https://github.com/sindresorhus/text-extensions#readme

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/redux/selectors.ts (hunk 2 above)
Comment (jmelnych) [reply src/redux/selectors.ts]: removed

**Review by topliceanurazvan — APPROVED** [review]
Great work 🙌🏻

**Review by devcatalin — APPROVED** [review]
LGTM and we can merge it, I'll follow-up on this next week to better define the `Files: Include` and `Files: Exclude` settings which we might not need anymore

**Pull request merged**